### ISSUE TITLE
Fixed potential vs. Computational hydrogen electrode methods

### ISSUE THREAD

**Issue by krl52** [shankar1729/jdftx#234]
Dear Shankar,
I would like to understand how to apply the target-mu parameter to calculate the H adsorption energy on copper. First I have calculated the energies of the H-Cu slab and clean Cu slab using the target-mu -(4.44-1.2)/27.2114 for a potential of -1.2 V vs SHE. The respective grand canonical free energies are -3195.51049 Eh and -3195.065238 Eh. Then I use the computational hydrogen electrode for G(H+ + e-) so 1/2 [E(H2) - TS] + e|U_SHE| - kT ln(10) pH where E(H2) is the DFT energy of gaseous hydrogen in vacuum (-1.177112292 Eh), TS is the entropy correction from rotational and translational movement (0.01483 Eh), e|U_SHE| = 1.2 eV, and -kT ln(10) pH ~ -0.00217 Eh ~ -0.059 eV for pH 1. Putting that all together, I have:

G(H on Cu) - G(Cu) - G(H+ + e-) 
= -3195.51049 - (-3195.065238 + 0.5 * (-1.177112292 -0.01483) + 0.0441 - 0.00217) 
= 0.1088 Hartree 
= 2.96 eV
That seems highly unfavorable at U = -1.2 V vs SHE and pH 1, so I'm wondering if there is something wrong with my interpretation of the method? 
Thank you very much for your help!

**Comment by shankar1729** [comment]
To be precise, in the grand canonical ensemble of electrons, you need G(H+), not G(H+ + e-), since the electron number is not constrained in the presence of an electron reservoir. To follow the logic through completely within the grand canonical picture, we have G(H+) = A(H+) - mu * 0 = A(H+) _at any potential_, since H+ has zero electrons.

Now to get the constant G(H+), we use the CHE. At mu = mu_SHE, H+ at pH = 1 are in equilibrium with H2, so that G(H+) = (1/2)G(H2) = (1/2) (E(H2) - TS - mu_SHE * 2) = (1/2) (E(H2) - TS) - mu_SHE = (1/2) (E(H2) - TS) + 4.44 eV.

So we have 4.44 eV in place of (1.2 + 0.059) eV in your expression for G(H+) above, which increases G(H+) by 3.18 eV and therefore reduces the adsorption energy to 2.96 - 3.18 = -0.22 eV.

In summary, there were two errors in your notes above:

1. You were using the logic for Helmholtz free energy, while comparing to grand free energies overall. You need to be consistent with the ensemble you use throughout.
2. The CHE already has pH = 1 in the H+ reference state, so there is no additional Nerstian correction (unless you need to go to a different pH). This was a smaller error, but still important.

Hope that helps!
Best,
Shankar

**Comment by krl52** [comment]
I see, that number makes a lot more sense, thank you! So if I compare to your 2017 paper on grand canonical DFT (eqns 31 and 32) for the H chemical potential you can derive something like
(E<sub>H</sub> - μN<sub>H</sub>) -1/2 (E<sub>H2=>2H</sub> + TS<sub>H2(g)</sub>) + (μ-μ<sub>SHE</sub> + eU<sub>H2(g)=>2H+</sub>) 
(E<sub>H</sub> - μN<sub>H</sub>) -1/2 (2E<sub>H</sub> - E<sub>H2</sub> + TS<sub>H2(g)</sub>) + (μ-μ<sub>SHE</sub> + eU<sub>H2(g)=>2H+</sub>)
and the E<sub>H</sub> and -1/2 (2 E<sub>H</sub>) terms cancel, as do the -μN<sub>H</sub> and + μ terms since N<sub>H</sub>=1, and eU<sub>H2(g)=>2H+</sub> = 0 by definition. Then you are left with

1/2 (E<sub>H2</sub> - TS<sub>H2(g)</sub>) -μ<sub>SHE</sub>
which is exactly what you wrote above. And then if you are to apply this to a molecule that adsorbs non-dissociatively (i.e. CO), you do not have the second and third terms and have only E<sub>CO</sub> - μN<sub>CO</sub>, where μ. is the same as the target-mu in the JDFTx input?

And lastly, If the computational hydrogen electrode is based on a pH of 1, does the correction for a different pH look like -kT ln(10)*(pH - 1)?

**Comment by shankar1729** [comment]
Yes, that's right for the molecule case. Note that it has nothing to do with dissociation; those extra terms in the H+ case were only because you were converting reference free energy of H2 to H+ using the SHE. When you don't convert like that, those terms are absent. So in your example, if hypothetically, CO splits to C and O when adsorbed on the surface, your expression is still the same as long as your reference state is gas-phase CO for your free energies.

For the pH correction, I didn't specifically check the sign, but yes it should be a Nernst equation shift of that form.

Best,
Shankar

**Comment by Lagoon23** [comment]
It is easy to determine the valence electrons N for metal atom. But I am confused how to determine the valence electrons for molecules such as CO CO2 and H2O. Just add up the valence electrons in each atom ? The valence electrons N for CO is 4+6=10. Am I doing right?

**Comment by shankar1729** [comment]
In all cases here, valence electrons refers to pseudo-valence electrons i.e. the number of electrons counted in each atom as valence by the pseudopotential. This is because the nElectrons included in the calculation of the grand free energy is based on all the electrons in the plane-wave calculation.

So in short, use the nElectrons from the calculation of the molecule using the same pseudopotentials, which you anyway need to do for the reference state for adsorption. For C, O and H, the number is almost certainly going to be 4, 6 and 1. However, in principle, you could use an all-electron pseudopotential with 1s electrons included explicitly which would make those numbers 6, 8 and 1. So it's best to use the number of electrons from the reference calculation.

Best,
Shankar

**Comment by ColinBundschu** [comment]
> The CHE already has pH = 1 in the H+ reference state, so there is no additional Nerstian correction (unless you need to go to a different pH). This was a smaller error, but still important.

I'm a little unclear on the terminology of the CHE, namely whether it refers to an SHE or an RHE, but to clarify this is using the RHE.

**Comment by krl52** [comment]
I thought that the CHE probably referred to the method developed in the 2004 Nørskov paper (https://doi.org/10.1021/jp047349j) The pH correction they use in step 5 is exactly the same factor that is used to convert the SHE potential to the RHE potential, so following their method I would think it would be 1/2 H2 - eU_SHE + kT ln(10) pH = 1/2 H2 - eU_RHE. They wrote that they use the SHE potential and apply the pH correction for pH different from 0 (steps 1 and 5) (hence the source of my confusion).
